### PR TITLE
Getting started with chat readme updates

### PR DESCRIPTION
**Pull request by yvprashanth**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,11 +7,13 @@ Before working on projects in this repo, we suggest taking time to follow [the p
 ## Solutions
 
 At the moment, these are the solutions in this repo:
-
-1) **[cloudformationTemplates/asyncCustomerChatUX](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/asyncCustomerChatUX)**
+ 1) **Get started with Chat** - **[CloudFormationTemplates/startChatContactAPI](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/startChatContactAPI)**
+    Use this solution to create a web widget with Amazon Connect Chat.
```
Comment (efh365) [inline README.md:11]: This solution does not create a widget. This one is just the API and has the customer manually adjust it.

Also, I intentionally put the async solution first as it requires no effort past the stack deployment and so I want customers to use that before the API one

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 2]
```diff
@@ -1,83 +1,173 @@
...
+Adding chat to your website is possible with a few easy steps. This solutions spins up an [Amazon API Gateway](https://aws.amazon.com/api-gateway/) endpoint that triggers an [AWS Lambda](https://aws.amazon.com/lambda/) function. This Lambda function invokes the [Amazon Connect](https://aws.amazon.com/connect/) Service [StartChatContact](https://docs.aws.amazon.com/en_pv/connect/latest/APIReference/API_StartChatContact.html) API and returns the result from that call. You could use the below pre built widget to get up and going. This solution is built on top of [Chat JS](https://github.com/amazon-connect/amazon-connect-chat-ui-examples.git).
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:3]: I find your modification of the last sentence more vague than what I have written

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 3]
```diff
@@ -1,83 +1,173 @@
...
+1. Create an Amazon Connect Instance by signing up [here](https://aws.amazon.com/connect/).
+You need an Amazon Connect instance to deploy the below [CloudFormation](https://aws.amazon.com/cloudformation/) template. You can use an existing one or create a new one by following our onboarding guide [here](https://docs.aws.amazon.com/connect/latest/adminguide/amazon-connect-get-started.html).
 
-If you are using an existing instance, you may need to make a few changes to your instance to enable Chat. Follow the steps [here](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/blob/master/README.md#enabling-chat-in-an-existing-amazon-connect-contact-center) to see what changes you need to make.
+    If you are using an existing instance, you may need to make a few changes to your instance to enable Chat. Follow the steps [here](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/blob/master/README.md#enabling-chat-in-an-existing-amazon-connect-contact-center) to see what changes you need to make.
 
-### Steps
+2. Once you create an Amazon Connect Instance, create a new contact flow, by picking **Sample queue customer** contact flow and **publish it**
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:15]: This doesn't make sense. How do you create a new flow by picking an existing flow?

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 4]
```diff
@@ -1,83 +1,173 @@
...
+    | ap-southeast-1 (Singapore) | [![Launch Stack](https://cdn.rawgit.com/buildkite/cloudformation-launch-stack-button-svg/master/launch-stack.svg)](https://ap-southeast-1.console.aws.amazon.com/cloudformation/home#/stacks/new?stackName=startChatContactAPI&templateURL=https://s3-ap-southeast-1.amazonaws.com/ap-southeast-1.start-chat-contact-api-cfn/cloudformation.yaml) |
 
-Note: you will want to show the widget only when there is a conversation in progress. If the widget is visible when there is no conversation, you will just see a loading spinner.
 
-1. In your website's html code, import the 'amazon-connect-chat-interface.js' file from this repo.
+4. Update the CloudFormation template with details from your connect instance.
+    - Fetch the Contact ID and Instance ID from your Connect instance by going to your contact flows and selecting **Sample queue customer** and clicking on **Show additional flow information** on top left.
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:33]: I don't like instructing customers to use a specific flow. If they are working on this stack they are more technically advanced and we can simply tell them to use the contact flow of their choosing. Right now you make it seem like it *has* to be the sample queue flow. This will lead to the customer having to retroactively go replace the contact flow id with one they want to use once everything is working.

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 5]
```diff
@@ -1,83 +1,173 @@
...
 
-1. In your website's html code, import the 'amazon-connect-chat-interface.js' file from this repo.
+4. Update the CloudFormation template with details from your connect instance.
+    - Fetch the Contact ID and Instance ID from your Connect instance by going to your contact flows and selecting **Sample queue customer** and clicking on **Show additional flow information** on top left. 
+    - `contactFlowId` \ `instanceId`: You can find the contact flow id and instance id when viewing a contact flow. For example, if the arn for your flow is 'arn:aws:connect:us-west-2:123456789012:instance/11111111-1111-1111-1111-111111111111/contact-flow/22222222-2222-2222-2222-222222222222', the contact flow id is '22222222-2222-2222-2222-222222222222' and instance id is 11111111-1111-1111-1111-111111111111
 
+        ![Contact Flow Id's](images/updateContactFlowIds.png)
+
+    - Go to your [CloudFormation stack](https://us-west-2.console.aws.amazon.com/cloudformation/home) by clicking on Stacks and then startChatContactAPI on the left hand side. Update the Contact Flow ID and the Instance ID in your cloud formation stack
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:38]: Why are you directing the customer to go to the CloudFormation console? They are already there by clicking on the launch stack button and you just told them to enter in the contact flow id and instance id right before this

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 6]
```diff
@@ -1,83 +1,173 @@
...
-3) Gather the contact flow ID and contact flow ID you want to use. If you don't have a contact flow, use the `Sample inbound flow` one in your instance that is created by default. You can find these IDs when viewing a contact flow. For example, if the arn for your flow is `arn:aws:connect:us-west-2:123456789012:instance/11111111-1111-1111-1111-111111111111/contact-flow/22222222-2222-2222-2222-222222222222`, the instance ID is `11111111-1111-1111-1111-111111111111` and the contact flow ID is `22222222-2222-2222-2222-222222222222`
-![contact flow ids diagram](images/contactFlowIds.png)
-4) Download the [compiled ChatJS source](https://github.com/amazon-connect/amazon-connect-chatjs/blob/master/dist/amazon-connect-chat.js) and save it locally.
-5) Create an `index.html` file in the same directory as the ChatJS source
-6) Set up your `index.html` file by importing ChatJS and creating the html elements you will use. These include an entry for the customer's name, a chat content input field, an area for the chat transcript to show up, and buttons to start the chat, get the transcript, send a typing event, and end the chat. 
+    <picture ?>
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:167]: Why did you remove the api gateway pic?

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 7]
```diff
@@ -1,83 +1,173 @@
...
     <script src="amazon-connect-chat-interface.js"></script>
+    <script src="https://code.jquery.com/jquery-3.4.1.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
+    ```
+
+2. Create a `div` with id of `root`. 
+    ```html
+    <div id="root">
+    </div>
     ```
+3. Get your API Gateway URL by going to your AWS Console and going to API Gateway product
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:57]: I don't think "API Gateway product" is a term that is used. "Get the API Gateway URL by going to API Gateway[link] in the AWS Console, selecting the API deployed by your CloudFormation stack, and navigating to the Stages page."

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 8]
```diff
@@ -1,83 +1,173 @@
...
 
-2. Initialize the Chat Interface on page load. Note: you need to update this to include the root id of the div where the customer chat widget will live.
+    ![API Gateway](images/apiGatewayUrl.png) 
 
-    ```js
-    $(document).ready((a) => {
-      connect.ChatInterface.init({
-        containerId: 'root' // This is the id of the container where you want the widget to reside
+4. Paste the below code as a script section in your html file and ***Update the 4 TODO fields***
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:61]: remove "as a script section". This is script and html

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 9]
```diff
@@ -1,83 +1,173 @@
...
+            console.log("There was an error: ");
+            console.log(error);
+        }
+    </script>
     ```
+5. Type your name and click on start chat. And then Go to your connect contact center https://<YourContactCenterName>.awsapps.com/connect/home and click on "Test Chat". Change your agent status as available to test the chat workflow
+
+    ![Test Chat](images/testChat.png) 
+
+6. That's it! Now you should be able to test AWS Connect Chat experience.
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:159]: "AWS Connect" is not our service name

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 10]
```diff
@@ -1,83 +1,173 @@
...
+            });
+        }
+
+        function failureHandler(error) {
+            console.log("There was an error: ");
+            console.log(error);
+        }
+    </script>
     ```
+5. Type your name and click on start chat. And then Go to your connect contact center https://<YourContactCenterName>.awsapps.com/connect/home and click on "Test Chat". Change your agent status as available to test the chat workflow
```
Comment (efh365) [inline cloudformationTemplates/startChatContactAPI/README.md:155]: I would not direct them to the test chat experience. That is to test the customer side. You should instead direct them to the CCP v2 so they can answer the chat that is in queue.

Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:155]: Yup. Agreed

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 9 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:159]: What is?

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 8 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:61]: It says "In your website's html code, import the 'amazon-connect-chat-interface.js' file from this repo. Also import jQuery"

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 7 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:57]: Can change it to Amazon API Gateway

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 6 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:167]: It exists in the pre built widget bullet #3

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 5 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:38]: In the previous command, we only tell them what the contact Flow ID and instance ID is and where to find them. Once they find them, this is to inform them on where they should update it within CloudFormation stack

File: cloudformationTemplates/startChatContactAPI/README.md (hunk 4 above)
Comment (yvprashanth) [reply cloudformationTemplates/startChatContactAPI/README.md:33]: We should make it as simple as possible for the customer to identify a beginner set of flows that they can learn / play with for chat. I've started the conversations with the right PM's to create Chat specific flows. This is the simplest flow I found that the customers can get started with.

File: README.md (hunk 1 above)
Comment (yvprashanth) [reply README.md:11]: Yup. I intentionally put the startChat one at the top is because, for customers who aren't as familiar with AWS understanding that architecture is another thing they have to learn. 
For customers who want to get started with a solution in their web app, with an index.html the startChat solution is much simpler.
